FWCore/Catalog/src/FileLocator.cc

```diff
@@ -41,23 +40,39 @@ namespace {
 
 }  // namespace
 
+namespace pt = boost::property_tree;
+
 namespace edm {
+
   FileLocator::FileLocator(std::string const& catUrl, unsigned iCatalog) : m_destination("any") {
-    init(catUrl, iCatalog);
+    init_trivialCatalog(catUrl, iCatalog);
+  }
 
-    // std::cout << m_protocols.size() << " protocols" << std::endl;
-    // std::cout << m_directRules[m_protocols[0]].size() << " rules" << std::endl;
+  FileLocator::FileLocator(edm::CatalogAttributes const& catAttr, unsigned iCatalog) : m_destination("any") {
+    init(catAttr, iCatalog);
   }
 
   FileLocator::~FileLocator() {}
 
-  std::string FileLocator::pfn(std::string const& ilfn) const { return convert(ilfn, m_directRules, true); }
-
-  std::string FileLocator::lfn(std::string const& ipfn) const { return convert(ipfn, m_inverseRules, false); }
+  std::string FileLocator::pfn(std::string const& ilfn, edm::CatalogType catType) const {
+    if (catType == edm::CatalogType::TrivialCatalog)
+      return convert(ilfn, m_directRules_trivialCatalog, true);
+    return convert(ilfn, m_directRules, true);
+  }
 
   std::string FileLocator::convert(std::string const& input, ProtocolRules const& rules, bool direct) const {
     std::string out = "";
-
+    //check if input is an authentic LFN
+    if (input.compare(0, 6, "/store") != 0)
```

Comment:
Could you make the `/store` a named constant? I.e. add after `kEmptyString` something along
```cpp
 constexpr char const* const kLFNPrefix = "/store";
```

Should the check require the LFN to start with `/store` or `/store/`? (@stlammel's suggestion in https://github.com/cms-sw/cmssw/pull/37278#issuecomment-1147908604 included the trailing slash)

Reply:
Done! I use ```/store/``` for valid LFN